File: backend/hackathon.API/Models/Notification.cs
using System.Text.Json;

namespace hackathon.API.Models;

public class Notification
{
    public Guid Id { get; set; }
    public Guid AudienceUserId { get; set; }
    public Guid? LinkedBookingId { get; set; }
    public Guid? LinkedSessionId { get; set; }
    public Guid? LinkedChargerId { get; set; }
    public NotificationTrigger TriggerEvent { get; set; }
    public NotificationChannel Channel { get; set; }
    public NotificationSeverity Severity { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public JsonDocument? Payload { get; set; }
    public NotificationDeliveryStatus DeliveryStatus { get; set; } = NotificationDeliveryStatus.Previewed;
    public bool ReadState { get; set; } = false;
    public string CorrelationId { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation
    public User AudienceUser { get; set; } = null!;
    public Booking? LinkedBooking { get; set; }
    public ChargingSession? LinkedSession { get; set; }
    public Charger? LinkedCharger { get; set; }
}

File: frontend/src/types/notification.ts
// Notification types — field names match api-contract.md §9.7 exactly

export type NotificationSeverity = 'Info' | 'Warning' | 'Critical';
export type NotificationChannel = 'InApp' | 'Email' | 'Teams';
export type DeliveryStatus = 'Sent' | 'Previewed' | 'Failed';

export type TriggerEvent =
  | 'BookingConfirmation'
  | 'SessionStartingSoon'
  | 'GracePeriodWarning'
  | 'SessionEndingSoon'
  | 'SessionEnded'
  | 'MoveVehicle'
  | 'SlotRelease'
  | 'AutoRelease'
  | 'AdminIntervention';

export interface Notification {
  id: string;
  triggerEvent: TriggerEvent;
  channel: NotificationChannel;
  severity: NotificationSeverity;
  title: string;
  body: string;
  readState: boolean;
  linkedBookingId: string | null;
  linkedSessionId: string | null;
  linkedChargerId: string | null;
  timestamp: string;
}

export interface NotificationAuditItem {
  id: string;
  audienceUserId: string;
  audienceUserDisplayName: string;
  triggerEvent: TriggerEvent;
  channel: NotificationChannel;
  severity: NotificationSeverity;
  title: string;
  body: string;
  payload: Record<string, unknown> | null;
  deliveryStatus: DeliveryStatus;
  readState: boolean;
  correlationId: string;
  linkedBookingId: string | null;
  linkedSessionId: string | null;
  linkedChargerId: string | null;
  timestamp: string;
}

export interface UnreadCountResponse {
  unreadCount: number;
}
